### ISSUE TITLE
Strange input image in android demo

### ISSUE THREAD

**Issue by barami98** [tensorflow/tensorflow#306]
I'm running android camera-based demo application. When I change SAVE_PREVIEW_BITMAP in TensorflowImageListener.java into true, I can save croppedBitmap. However, this saved image is all green with little detail of image.

![preview](https://cloud.githubusercontent.com/assets/13412657/11295381/69658806-8fae-11e5-8db7-9943d1f5a195.png)

When the camera is connected, YUV_420_888 format is used, Then plane #0 is always Y, plane #1 is always U (Cb), and plane #2 is always V (Cr).

```
imageReader =
            ImageReader.newInstance(
                largest.getWidth(), largest.getHeight(), ImageFormat.YUV_420_888, /*maxImages*/ 2);
```

However in TensorflowImageListener.java only 0, 2 planes are used.

```
final int[] planeOrder = {0, 2};
for (int i = 0; i < planeOrder.length; ++i) {
  final Plane plane = planes[planeOrder[i]];
  final ByteBuffer buffer = plane.getBuffer();
  buffer.rewind();
  final int readAmount = buffer.remaining();
  buffer.get(yuvBytes, position, readAmount);
  position += readAmount;
}
```

By the way I'm using LG G2 and Android version is 5.0.1

**Comment by andrewharp** [comment]
Hi,
This behavior should be fixed in with commit https://github.com/tensorflow/tensorflow/commit/854f49bd43588c062b046384f239f64a3d819702

An upcoming change will push all of the colorspace conversion into native code as well.

**Comment by vrv** [comment]
color space conversion change in bf6b536bde7d8060c489b51fedb58968b8cbfd7c

**Comment by barami98** [comment]
I found that Y channel is OK but U, V channels are all zero because of low android version.
It was fixed in Android 5.1.1

https://code.google.com/p/android/issues/detail?id=81984

**Comment by andrewharp** [comment]
I see, thanks for the update. Does this occur at all preview resolutions?
